lib/ansible/modules/cloud/google/gcp_healthcheck.py

```diff
@@ -0,0 +1,464 @@
+    :type params:  ``dict``
+
+    :return: Tuple with changed status and response dict
+    :rtype: ``tuple`` in the format of (bool, dict)
+    """
+    gcp_dict = _build_healthcheck_dict(params)
+    try:
+        resource, _ = _get_req_resource(client, resource_type)
+        args = {'project': project_id, 'body': gcp_dict}
+        req = resource.insert(**args)
+        return_data = GCPUtils.execute_api_client_req(req, client, raw=False)
+        if not return_data:
+            return_data = get_healthcheck(client,
+                                          name=params['healthcheck_name'],
+                                          project_id=project_id)
+        return (True, return_data)
+    except:
+        raise
+
+
+def delete_healthcheck(client, name, project_id, resource_type='HTTP'):
+    """
+    Delete a Healthcheck.
+
+    :param client: An initialized GCE Compute Disover resource.
+    :type client:  :class: `googleapiclient.discovery.Resource`
+
+    :param name: Name of the Url Map.
+    :type name:  ``str``
+
+    :param project_id: The GCP project ID.
+    :type project_id:  ``str``
+
+    :return: Tuple with changed status and response dict
+    :rtype: ``tuple`` in the format of (bool, dict)
+    """
+    try:
+        resource, entity_name = _get_req_resource(client, resource_type)
+        args = {'project': project_id, entity_name: name}
+        req = resource.delete(**args)
+        return_data = GCPUtils.execute_api_client_req(req, client)
+        return (True, return_data)
+    except:
+        raise
+
+
+def update_healthcheck(client, healthcheck, params, name, project_id,
+                       resource_type='HTTP'):
+    """
+    Update a Healthcheck.
+
+    If the healthcheck has not changed, the update will not occur.
+
+    :param client: An initialized GCE Compute Disovery resource.
+    :type client:  :class: `googleapiclient.discovery.Resource`
+
+    :param healthcheck: Name of the Url Map.
+    :type healthcheck:  ``dict``
+
+    :param params: Dictionary of arguments from AnsibleModule.
+    :type params:  ``dict``
+
+    :param name: Name of the Url Map.
+    :type name:  ``str``
+
+    :param project_id: The GCP project ID.
+    :type project_id:  ``str``
+
+    :return: Tuple with changed status and response dict
+    :rtype: ``tuple`` in the format of (bool, dict)
+    """
+    gcp_dict = _build_healthcheck_dict(params)
+    ans = GCPUtils.are_params_equal(healthcheck, gcp_dict)
+    if ans:
+        return (False, 'no update necessary')
+
+    try:
+        resource, entity_name = _get_req_resource(client, resource_type)
+        args = {'project': project_id, entity_name: name, 'body': gcp_dict}
+        req = resource.update(**args)
+        return_data = GCPUtils.execute_api_client_req(
+            req, client=client, raw=False)
+        return (True, return_data)
+    except:
+        raise
+
+
+def main():
+    module = AnsibleModule(argument_spec=dict(
```

Comment:
Anywhere you've set a default in `argument_spec` please ensure that
1) You do `default: $value` 
2) Remove the default from the `description:`

See http://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html#documentation-block for mor einfo